prov/gni/include/gnix_freelist.h

```diff
@@ -123,6 +126,12 @@ static inline int _gnix_fl_alloc(struct dlist_entry **e, struct gnix_freelist *f
         fastlock_acquire(&fl->lock);
 
     if (dlist_empty(&fl->freelist)) {
+
+        if (fl->refill_size == 0) {
+            ret = -FI_ECANCELED;
```

Comment:
Okay, I guess I can be convinced to return -FI_ECANCELED.  Let's go with that.  I agree with jswaro that a GNIX_WARN might be good when -FI_ECANCELED is going to be returned.